compiler/test/dotty/tools/dotc/CompilationTests.scala

```diff
@@ -229,9 +229,11 @@ class CompilationTests {
     ).checkCompile()
 
 
-  /** The purpose of this test is three-fold, being able to compile dotty
-   *  bootstrapped, and making sure that TASTY can link against a compiled
-   *  version of Dotty, and compiling the compiler using the SemanticDB generation
+  /** This test serves several purposes:
+   *   - being able to compile dotty bootstrapped,
+   *   - making sure that TASTY can link against a compiled version of Dotty,
+   *   - compiling the compiler using the SemanticDB generation
+   *   - compiling the compiler under -Yrecheck mode.
```

Comment:
Can -Yrecheck be thought of as a more extensive of -Ycheck or do they end up checking different things?

Reply:
Quite different things. -Yrecheck is only for testing special recheck phases. Right now there is only one such phase planned, and that is the capture checker. But others might follow.